src/com/boxboat/jenkins/library/Utils.groovy

```diff
@@ -65,6 +65,76 @@ class Utils implements Serializable {
         return format
     }
 
+    static def generateUUID(){
+        return UUID.randomUUID().toString()
+    }
+
+    static def makeTmpFile(){
+        def tmpFile = "./tmp-${generateUUID()}"
+        Config.pipeline.sh """
+            set +x
+            touch ${tmpFile}
+        """
+        return tmpFile
+    }
+
+    static void withTmpFile(Map configMap=[:], Closure closure){
```

Comment:
`configMap` does not make sense here in a global utility function.  The `configMap` seems to expect a strict key, and would be better represented as an Object so that the contract is well defined.  Since the environment variable setting is a specific implementation, that should be part of the calling function